lib/src/test/java/uk/gov/di/ipv/cri/passport/library/service/PassportSessionServiceTest.java

```diff
@@ -63,10 +63,42 @@ void shouldCreateSessionItem() throws ParseException {
 
         ArgumentCaptor<PassportSessionItem> passportSessionItemArgumentCaptor =
                 ArgumentCaptor.forClass(PassportSessionItem.class);
-        verify(dataStore).create(passportSessionItemArgumentCaptor.capture());
+        verify(mockDataStore).create(passportSessionItemArgumentCaptor.capture());
         assertNotNull(passportSessionItemArgumentCaptor.getValue().getCreationDateTime());
         assertEquals(
                 passportSessionItemArgumentCaptor.getValue().getPassportSessionId(),
                 passportSessionID);
     }
+
+    @Test
+    void shouldUpdateLatestDcsResponseResourceId() {
+        String passportSessionID = SecureTokenHelper.generate();
+        String latestDcsResponseResourceId = "test";
+
+        when(mockDataStore.getItem(passportSessionID)).thenReturn(new PassportSessionItem());
+        underTest.setLatestDcsResponseResourceId(passportSessionID, latestDcsResponseResourceId);
+
+        ArgumentCaptor<PassportSessionItem> passportSessionItemArgumentCaptor =
+                ArgumentCaptor.forClass(PassportSessionItem.class);
+        verify(mockDataStore).update(passportSessionItemArgumentCaptor.capture());
+        assertEquals(
+                passportSessionItemArgumentCaptor.getValue().getLatestDcsResponseResourceId(),
+                latestDcsResponseResourceId);
+    }
+
+    @Test
+    void shouldIncrementAttemptCount() {
+        String passportSessionID = SecureTokenHelper.generate();
+
+        PassportSessionItem passportSessionItem = new PassportSessionItem();
+        passportSessionItem.setAttemptCount(1);
+
+        when(mockDataStore.getItem(passportSessionID)).thenReturn(passportSessionItem);
+        underTest.incrementAttemptCount(passportSessionID);
+
+        ArgumentCaptor<PassportSessionItem> passportSessionItemArgumentCaptor =
+                ArgumentCaptor.forClass(PassportSessionItem.class);
+        verify(mockDataStore).update(passportSessionItemArgumentCaptor.capture());
+        assertEquals(passportSessionItemArgumentCaptor.getValue().getAttemptCount(), 2);
```

Comment:
There is a sonar code smell here. Think the values in the assertEquals just need swapping round
```suggestion
        assertEquals(2, passportSessionItemArgumentCaptor.getValue().getAttemptCount());
```

Reply:
Addressed in new PR.